pulp_container/app/viewsets.py

```diff
@@ -459,10 +462,11 @@ def untag(self, request, pk):
         Create a task which is responsible for untagging an image.
         """
         repository = self.get_object()
-        request.data["repository"] = repository
+        repository_version = repository.latest_version()
```

Comment:
Why don't we add the repository to the serializer context instead?
This kind of implies that we could tag something from a baseversion, which we cannot. (With retained_version=1, the task may see a completely different single version of this repo as the serializer.)

Reply:
We cannot pass objects to a serializer directly in this case. There are two options:
1. Pass the HREF or PK of a repository as a string to the serializer (we will need to add a new serializer field that will be visible in the REST API documentation),
2. Pass a repository object to a serializer with the context argument (https://www.django-rest-framework.org/api-guide/serializers/#including-extra-context).

The second option looks better to me. Also, we are doing this everywhere where we are dealing with repository operations.

Yes, I should rename `repository_version` back to `latest_version`.

Reply:
I agree for option 2. My point is just we pass the repository (not the version) to the task, therefore we should use it to validate too.
So we would pass `context={"request": request, "repository": repository}`